### PR TITLE
DATACASS-188 - now able to supply options to table via cassandra admin otemplate

### PR DESCRIPTION
**Pull request by Mikhail2048**
Implements #359
@mp911de, can you please check?)

### CODE REVIEW

File: spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraAdminOperations.java [hunk 1]
```diff
@@ -46,15 +47,31 @@ public interface CassandraAdminOperations extends CassandraOperations {
...
 	 *
+	 * @deprecated Deprecated in favor of {@link #createTable(boolean, Class, Map)}, since we can infer the table name from
+	 *             passed entity class
 	 * @param ifNotExists If true, will only create the table if it doesn't exist, else the create operation will be
 	 *          ignored.
 	 * @param tableName The name of the table.
 	 * @param entityClass The class whose fields determine the columns created.
 	 * @param optionsByName Table options, given by the string option name and the appropriate option value.
 	 */
+	@Deprecated(forRemoval = true)
```
Comment (mp911de) [inline spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraAdminOperations.java:58]: We do not intent deprecating this method as a table can be created from the given `Class` using a different table name.

File: spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraAdminTemplate.java [hunk 2]
```diff
@@ -107,11 +110,27 @@ public SchemaFactory getSchemaFactory() {
...
+	@Override
+	public void createTable(boolean ifNotExists, Class<?> entityClass, Map<String, Object> optionsByName) {
 		CassandraPersistentEntity<?> entity = getConverter().getMappingContext().getRequiredPersistentEntity(entityClass);
 
 		CreateTableSpecification createTableSpecification = this.schemaFactory
-				.getCreateTableSpecificationFor(entity, tableName).ifNotExists(ifNotExists);
+				.getCreateTableSpecificationFor(entity, entity.getTableName())
+				.ifNotExists(ifNotExists);
+
+		if (!CollectionUtils.isEmpty(optionsByName)) {
```
Comment (mp911de) [inline spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraAdminTemplate.java:124]: This is actually a bugfix. Can you please split this pull request into two PR's, one for introducing an overload without accepting a table name and the other one to consider options?

File: spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraTemplate.java [hunk 3]
```diff
@@ -961,7 +961,7 @@ protected <E extends CassandraMappingEvent<T>, T> void maybeEmitEvent(Supplier<E
 	protected <T> T maybeCallBeforeConvert(T object, CqlIdentifier tableName) {
 
 		if (null != entityCallbacks) {
-			return (T) entityCallbacks.callback(BeforeConvertCallback.class, object, tableName);
+			return entityCallbacks.callback(BeforeConvertCallback.class, object, tableName);
```
Comment (mp911de) [inline spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraTemplate.java:964]: These should go into a separate commit as well to separate polishing from the actual change.

**Review by mp911de — CHANGES_REQUESTED** [review]
There are quite some unrelated changes within a single commit. Please split at least the bugfix from the newly introduced method. The polishing changes should be part of a separate commit, can be either PR.

**Comment by Mikhail2048** [comment]
@mp911de I have removed new method from this PR. Also I have introduced 2 commits - one with polishing, another with actual bugfix. Can you please re-review?
 
P.S: I will introduce new method once this changes will occur in the _main_ branch

**Comment by Mikhail2048** [comment]
Mark @mp911de, when you will be able to rereview this PR?) I have already created a separate brunch for new overloaded method, but I am waiting for these changes to occur in main brunch to create new PR as we discussed

**Pull request closed without merging**